components/footer-link/footer-link.js

```diff
@@ -0,0 +1,29 @@
+const template = document.createElement("template");
+template.innerHTML = `
+  <link rel="stylesheet" href="./components/footer-link/footer-link.css">
+  <ul class="footer-link">
+    <li>
+      <a id="link" href="#">FAQ</a>
+    </li>
+  </ul>
+`;
+
+class FooterLinkComponent extends HTMLElement {
+  constructor() {
+    super();
+    const shadow = this.attachShadow({ mode: "open" });
+    shadow.appendChild(template.content.cloneNode(true));
+  }
+
+  static get observedAttributes() {
+    return ["href"];
+  }
+
+  attributeChangedCallback(name, oldValue, newValue) {
+    if (name === "href") {
+      this.shadowRoot.getElementById("link").setAttribute("href", newValue);
+    }
+  }
+}
```

Comment:
마이너한 부분인 것 같긴한데, attributeChangedCallback 메소드 내, href를 업데이트 시켜주는 로직을 분리해주는 것도 유지보수와 코드 가독성 측면에서 좋을 것 같습니다 :)
추가로, 매번 this.shadowRoot.getElementById("link")를 호출해주기보다, 초기에 인스턴스 변수를 하나 만들어 한번만 link element에 저장해주는 것도 DOM 접근을 최소화할 수 있는 방법일 것 같습니다!
DOM 조작 최적화와 관련해서는 [이 블로그](https://medium.com/@sudheer.gowrigari/optimizing-rendering-performance-in-web-components-a-comprehensive-guide-a507ba4afe19)를 참고해보셔도 좋을 것 같습니다. 핵심은, 효율적인 성능과 사용자 경험을 위함이라고 생각해주시면 됩니다!
```suggestion
// 초기에 link element 인스턴스 변수 만들어서 저장하고 초기 값 설정
constructor() {
    this.linkElement = shadow.getElementById("link");
    this.updateLink(this.getAttribute("href"));
}

// 기존 attributeChangedCallback 업데이트
attributeChangedCallback(name, oldValue, newValue) {
    if (name === "href") {
        this.updateLinkHref(newValue);
    }
}

// link를 update시켜주는 로직 분리
updateLink(href) {
    this.linkElement.setAttribute("href", href);
}
```

Reply:
추천해주신대로 코드 변경해봤어요! 역시 여러 동작을 한 줄에 넣는건 가독성을 해치는 것 같습니다. DOM 관련 내용도 잘 이해하도록 한 번 읽어볼게요! 감사드려요.